Marlin/src/inc/Conditionals_post.h

```diff
@@ -2907,6 +2907,16 @@
   #define HAS_ROTARY_ENCODER 1
 #endif
 
+#if NUM_SERIAL == 1 && defined(SERIAL_PORT_2)
+  // There's a bug here while platform IO is computing the BUILD_FEATURES for its scripts. It's including
+  // common-dependencies.h that (force) defines NUM_SERIAL to 1
+  // In turn, because NUM_SERIAL is set to 1, the code below undefine MEATPACK
+  // if it's defined from MEATPACK_ON_SERIAL_PORT_2 and it's not build anymore (it fails at link step)
```

Comment:
So we should move the definition of NUM_SERIAL out of the hal. Is it possible?

Reply:
just move 
```
#ifdef SERIAL_PORT_2
  #define NUM_SERIAL 2
#else
  #define NUM_SERIAL 1
#endif
```
to `Conditionals_post.h`
And remove `#define NUM_SERIAL 1` from `buildroot/share/PlatformIO/scripts/common-dependencies.h`

Reply:
Some HAL define serial port by themselves (like LPC176x that's defining the LCD_SERIAL_PORT and MMU_SERIAL_PORT). I'm not sure if it's counted in the NUM_SERIAL macro.

We could also have a single `#define SERIAL_PORTS { -1, 1, 2, 3 }` (leading to `constexpr int SerialPorts[] = { -1, 1, 2, 3 }` and then just `#define NUM_SERIAL COUNT(SerialPorts)`. 
However, constructing the serial chain from such a list will be a template monster... Or not. Maybe it's worth investigating.  

It's clear that it'll probably be easier than what we have currently with all this "#ifdef SERIAL_IS_SOMETHING #else if something else" chain.

Reply:
`NUM_SERIAL`, as I remember, is only related to the presence of SERIAL_1 and SERIAL_2 ... the others serial have different meaning and handling ...

Reply:
@thinkyhead can we move the define of `NUM_SERIAL` to `Conditionals_post.h`?

Reply:
I'm not available this WE for development. On monday, I'll change the way you said and check it also fix the issue.